File: app/src/main/java/com/shevelev/comics_viewer/dal/dto/Page.java
package com.shevelev.comics_viewer.dal.dto;

import com.shevelev.comics_viewer.comics_workers.PreviewCreator;
import com.shevelev.comics_viewer.dal.entities.DbPage;

/**
 * One page of comics
 */
public class Page
{
    public long id;

    /** Filename without path */
    public String fileName;

    public int order;

    public boolean isLeftTopCornerDark;

    public boolean isLeftBottomCornerDark;

    public boolean isRightTopCornerDark;

    public boolean isRightBottomCornerDark;

    public Page()
    {
    }

    public Page(DbPage dbPage)
    {
        fileName = dbPage.fileName;
        order = dbPage.order;
        id = dbPage.getId();

        isLeftBottomCornerDark = dbPage.isLeftBottomCornerDark;
        isLeftTopCornerDark = dbPage.isLeftTopCornerDark;
        isRightBottomCornerDark = dbPage.isRightBottomCornerDark;
        isRightTopCornerDark = dbPage.isRightTopCornerDark;
    }

    public String getPreviewFileName()
    {
        return PreviewCreator.getPreviewFileName(fileName);
    }
}

File: app/src/main/java/com/shevelev/comics_viewer/dal/dto/Page.kt
package com.shevelev.comics_viewer.dal.dto

import com.shevelev.comics_viewer.comics_workers.PreviewCreator.Companion.getPreviewFileName
import com.shevelev.comics_viewer.dal.entities.DbPage

/**
 * One page of comics
 */
class Page {
    var id: Long = 0
    /** Filename without path  */
    var fileName: String? = null
    var order = 0
    var isLeftTopCornerDark = false
    var isLeftBottomCornerDark = false
    var isRightTopCornerDark = false
    var isRightBottomCornerDark = false

    constructor() {}
    constructor(dbPage: DbPage) {
        fileName = dbPage.fileName
        order = dbPage.order
        id = dbPage.id
        isLeftBottomCornerDark = dbPage.isLeftBottomCornerDark
        isLeftTopCornerDark = dbPage.isLeftTopCornerDark
        isRightBottomCornerDark = dbPage.isRightBottomCornerDark
        isRightTopCornerDark = dbPage.isRightTopCornerDark
    }

    val previewFileName: String?
        get() = getPreviewFileName(fileName)
}